jina/serve/networking.py

```diff
@@ -28,11 +31,12 @@ class ReplicaList:
     Maintains a list of connections to replicas and uses round robin for selecting a replica
     """
 
-    def __init__(self):
+    def __init__(self, summary):
         self._connections = []
         self._address_to_connection_idx = {}
         self._address_to_channel = {}
         self._rr_counter = 0
+        self.summary = summary
```

Comment:
why are u adding this, shouldn't this be part of another PR?

Reply:
the metrics PR was merged already, but I am changing the ConnectionPool a bit and thus I need the prometheus summary object elswhere.
So I am passing it done to the new object introduced by this PR